jerry-core/parser/js/parser.cpp

```diff
@@ -2881,11 +2881,23 @@ preparse_scope (bool is_global)
   bool is_ref_eval_identifier = false;
   bool is_use_strict = false;
 
-  if (token_is (TOK_STRING) && lit_literal_equal_type_cstr (lit_get_literal_by_cp (token_data_as_lit_cp ()),
-                                                            "use strict"))
+  /*
+   * Check Directive Prologue for Use Strict directive (see ECMA-262 5.1 section 14.1)
+   */
+  while (token_is (TOK_STRING))
   {
-    scopes_tree_set_strict_mode (STACK_TOP (scopes), true);
-    is_use_strict = true;
+    if (lit_literal_equal_type_cstr (lit_get_literal_by_cp (token_data_as_lit_cp ()), "use strict")
+        && lexer_is_no_escape_sequences_in_token_string (tok))
```

Comment:
I think it could be faster if you swap the part of the conditions. The last check seems much simpler for me and if it fails, the strings shouldn't be compared.